packages/protocol/script/deploy.s.sol

```diff
@@ -0,0 +1,236 @@
+        console.log("AddressManagerProxy:", addressManagerProxy);
+
+        // TaikoToken
+        TaikoToken taikoToken = new TaikoToken();
+
+        address[] memory premintRecipients = new address[](1);
+        uint256[] memory premintAmounts = new uint256[](1);
+        premintRecipients[0] = taikoTokenPremintRecipient;
+        premintAmounts[0] = taikoTokenPremintAmount;
+
+        address taikoTokenProxy = deployUupsProxy(
+            address(taikoToken),
+            proxyAdmin,
+            bytes.concat(
+                taikoToken.init.selector,
+                abi.encode(
+                    addressManagerProxy,
+                    "Taiko Token",
+                    "TKO",
+                    premintRecipients,
+                    premintAmounts
+                )
+            )
+        );
+        console.log("TaikoToken:", address(taikoToken));
+        console.log("TaikoTokenProxy:", taikoTokenProxy);
+
+        // HorseToken && BullToken
+        FreeMintERC20 horseToken = new FreeMintERC20("Horse Token", "HORSE");
+        MayFailFreeMintERC20 bullToken = new MayFailFreeMintERC20(
+            "Bull Token",
+            "BLL"
+        );
+        console.log("HorseToken:", address(horseToken));
+        console.log("BullToken:", address(bullToken));
+
+        // TaikoL1
+        TaikoL1 taikoL1 = new TaikoL1();
+        uint64 feeBase = 10 ** 18;
+        address taikoL1Proxy = deployUupsProxy(
+            address(taikoL1),
+            proxyAdmin,
+            bytes.concat(
+                taikoL1.init.selector,
+                abi.encode(addressManagerProxy, l2GensisHash, feeBase)
+            )
+        );
+        console.log("TaikoL1:", address(taikoL1));
+        console.log("TaikoL1Proxy:", taikoL1Proxy);
+
+        // Used by TaikoToken
+        AddressManager(addressManagerProxy).setAddress(
+            string.concat(l1ChainId, ".proto_broker"),
+            address(taikoL1Proxy)
+        );
+        // Used by LibBridgeRead
+        AddressManager(addressManagerProxy).setAddress(
+            string.concat(l1ChainId, ".taiko"),
+            address(taikoL1Proxy)
+        );
+
+        // Bridge
+        Bridge bridge = new Bridge();
+        address bridgeProxy = deployUupsProxy(
+            address(bridge),
+            proxyAdmin,
+            bytes.concat(bridge.init.selector, abi.encode(addressManagerProxy))
+        );
+        console.log("Bridge:", address(bridge));
+        console.log("BridgeProxy:", bridgeProxy);
+
+        TokenVault tokenVault = new TokenVault();
+        address tokenVaultProxy = deployUupsProxy(
+            address(tokenVault),
+            proxyAdmin,
+            bytes.concat(
+                tokenVault.init.selector,
+                abi.encode(addressManagerProxy)
+            )
+        );
+        console.log("TokenVault:", address(tokenVault));
+        console.log("TokenVaultProxy:", tokenVaultProxy);
```

Comment:
I think we can extra a function for this pattern:

```
TokenVault tokenVault = new TokenVault();
        address tokenVaultProxy = deployUupsProxy(
            address(tokenVault),
            proxyAdmin,
            bytes.concat(
                tokenVault.init.selector,
                abi.encode(addressManagerProxy)
            )
        );
        console.log("TokenVault:", address(tokenVault));
        console.log("TokenVaultProxy:", tokenVaultProxy);
```

Reply:
changed